dll/win32/comctl32/listview.c

```diff
@@ -3254,7 +3254,18 @@ static RANGES ranges_clone(RANGES ranges)
 {
     RANGES clone;
     INT i;
-	   
+
+#ifdef __REACTOS__
+    if (!ranges || !ranges->hdpa)
+    {
+        /* 
+         * If a ExplorerBand tree rename operation is completed by left-clicking in
+         * DefView, the navigation to the newly named item causes the ListView in DefView
+         * to call LISTVIEW_DeselectAllSkipItems during ListView destruction.
```

Comment:
Is this a behaviour that is "naturally" expected (e.g. on Windows), or a current limitation (or bug?) in ReactOS' shell?
(Looks like a bug to me because here you had to hack into comctl32 and not in browseui or shell32...)
If the problem described still happens in ReactOS even if you replace its comctl32.dll with Windows 2003's one, this would point to that problem. But then, I would strongly suggest you don't insert a workaround in comctl32 but fix (or workaround) instead in shell32 or browseui.

Reply:
I can work around the problem with PostMessage in browseui but that feels a bit hacky when neither the crash nor the component that own the control is in browseui.

Reply:
You can trigger the crash on older builds as well (tested on a several months old build) and it really has nothing to do with the context menu, just clicking to enter edit mode and you can trigger it.